sdk/metric/export/data.go

```diff
@@ -0,0 +1,90 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: NOTE this is a temporary space, it may be moved following the
+// discussion of #2813, or #2841
+
+package export // import "go.opentelemetry.io/otel/sdk/metric/export"
+
+import (
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+type (
+	// Metrics is the result of a single collection.
+	//
+	// This struct supports re-use of the nested memory structure
+	// in its Scopes slice such that repeated calls Produce will
+	// not reallocate the same quantity of memory again and again.
+	//
+	// To re-use the memory from a previous Metrics value, pass a
+	// pointer to the former result to Produce(). This is safe for
+	// push-based exporters that perform sequential collection.
+	Metrics struct {
+		// Resource is the MeterProvider's configured Resource.
+		Resource *resource.Resource
+
+		// Scopes is a slice of metric data, one per Meter.
+		Scopes []Scope
+	}
+
+	// Scope is the result of a single collection for a single Meter.
+	//
+	// See the comments on Metrics about re-use of slices in this struct.
+	Scope struct {
+		// Library describes the instrumentation scope.
+		Library instrumentation.Library
+
+		// Instruments is a slice of metric data, one per Instrument
+		// in the scope.
+		Instruments []Instrument
+	}
+
+	// Instrument is the result of a single collection for a single Instrument.
+	//
+	// See the comments on Metrics about re-use of slices in this struct.
+	Instrument struct {
+		// Descriptor describes an instrument created through a View,
+		// including name, unit, description, instrument and number kinds.
+		// TODO Define sdkinsturment package #2813
+		// Descriptor sdkinstrument.Descriptor
+
+		// Points is a slice of metric data, one per attribute.Set value.
+		Points []Point
+	}
+
+	// Point is a timeseries data point resulting from a single collection.
+	Point struct {
+		// Attributes are the coordinates of this series.
+		Attributes attribute.Set
+
+		// Aggregation determines the kind of data point
+		// recorded in this series.
+		// TODO: Define aggregation types #2827, #2828
+		// Aggregation aggregation.Aggregation
+
+		// Start indicates the start of the collection
+		// interval reflected in this series, which is set
+		// according to the configured temporality.
+		Start time.Time
+
+		// End indicates the moment at which the collection
+		// was performed.
+		End time.Time
+	}
+)
```

Comment:
This includes a lot of structure and implementation details that is not directly related to adding a `Reader` interface. Can we instead add a minimal struct here?

```suggestion
// Metrics is the result of a single collection.
type Metrics struct { /* TODO: implement */ }
```

Reply:
Agree -- we could leave this as `struct Metrics { /* TODO */ }`.

For what it's worth, I recommend renaming this package from `export` to `data`. The "export" prefix has a history and `sdk/trace` has no equivalent. As I read type names `data.Metrics`, `data.Scope`, `data.Instrument`, and `data.Point`, I'm reminded that these are plain-old data objects.

Reply:
I've reduced this to just an empty struct. 

Can we defer the decision of if this should be `export.Metrics` or `data.Metrics` until we fill out this struct?  I made #2889 to track this issue.